### PR TITLE
Basic Tests

### PR DESCRIPTION
**Pull request by gauntface**
cc @addyosmani @surma 

Nothing exciting but it's a start to testing this stuff.

Will add Travis in a separate PR if this looks ok and expend these to cover error cases, then I'll add in support for testing watch scenarios.

### CODE REVIEW

File: test/task-checker.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import * as sass from '../src/wsk-tasks/sass.js';
+
+describe('This should enforce any rules we want to exist for our tasks', () => {
+  // Must clean up the output path
+  let fs = require('fs');
+  let rimraf = require('rimraf');
+
+  const VALID_TEST_FILES = 'test/data/valid-files';
+  const TEST_OUTPUT_PATH = 'test/output';
```
Comment (surma) [inline test/task-checker.js]: Add this to `.gitignore`?

File: test/task-checker.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  let tasksToTest = [
+    {taskName: 'sass', task: sass}
+  ];
+
+  beforeEach(done => {
```
Comment (surma) [inline test/task-checker.js]: Stylistic Q: why not single line? `done => rimraf(TEST_OUTPUT_PATH, done)`?

File: test/task-checker.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    });
+
+    it('should build cleanly', done => {
+      GLOBAL.config = {
+        env: 'prod',
+        src: VALID_TEST_FILES,
+        dest: TEST_OUTPUT_PATH
+      };
+
+      task.build().on('end', () => {
```
Comment (surma) [inline test/task-checker.js]: Where are we saying that the modules should put their output to `TEST_OUTPUT_PATH`? Can’t see it :cry:

Comment (gauntface) [inline test/task-checker.js]: The `GLOBAL.config` is what the tasks will read where to look and put files.

TEST_OUTPUT_PATH is defined on line 34.

Is that your question?

File: test/task-checker.js (hunk 2 above)
Comment (gauntface) [inline test/task-checker.js]: I'm an old style dev with old style practices?

Changed.

File: test/task-checker.js (hunk 1 above)
Comment (gauntface) [inline test/task-checker.js]: Done.Good call.

File: test/task-checker.js (hunk 3 above)
Comment (surma) [inline test/task-checker.js]: lol

**Pull request merged**